File: VB/ServerSide/Controllers/WebDocumentViewerController.vb
Imports DevExpress.Web.Mvc.Controllers
Imports DevExpress.XtraReports.Web.Extensions
Imports System.Linq
Imports System.Web.Mvc

Namespace ServerSide.Controllers
    Public Class WebDocumentViewerController
        Inherits WebDocumentViewerApiController

        Public Overrides Function Invoke() As ActionResult
            Dim result = MyBase.Invoke()
            ' Allow cross-domain requests.
            Response.AppendHeader("Access-Control-Allow-Origin", "*")
            Return result
        End Function

        <HttpPost> _
        Public Function GetReports() As ActionResult
            Response.AppendHeader("Access-Control-Allow-Origin", "*")
            Dim result = New JsonResult With {.Data = ReportStorageWebService.GetUrls().ToArray()}
            Return result
        End Function
    End Class
End Namespace

File: CS/ServerSide/Controllers/WebDocumentViewerController.cs
using System.Linq;
using DevExpress.Web.Mvc.Controllers;
using System.Web.Mvc;
using DevExpress.XtraReports.Web.Extensions;

namespace ServerSide.Controllers
{
    public class WebDocumentViewerController : WebDocumentViewerApiControllerBase {
        public override ActionResult Invoke() {
            var result = base.Invoke();
            // Allow cross-domain requests.
            Response.AppendHeader("Access-Control-Allow-Origin", "*");
            return result;
        }

        [HttpPost]
        public ActionResult GetReports() {
            Response.AppendHeader("Access-Control-Allow-Origin", "*");
            var result = new JsonResult {
                Data = ReportStorageWebService.GetUrls().ToArray()
            };
            return result;
        }
    }
}